specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,249 @@
+// application, the button can be updated to appear selected when the
+// dialog is opened.
+webView.CoreWebView2.IsDefaultDownloadDialogOpenChanged += delegate (
+    object sender, object e)
+{
+    if (webView.CoreWebView2.IsDefaultDownloadDialogOpen)
+    {
+        // Here the developer can make updates in response to
+        // the default download dialog opening.
+    }
+    else
+    {
+        // Here the developer can make updates in response to the
+        // default download dialog closing.
+    }
+};
+
+void ToggleDownloadDialogCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    if (webView.CoreWebView2.IsDefaultDownloadDialogOpen)
+    {
+        webView.CoreWebView2.CloseDefaultDownloadDialog();
+    }
+    else
+    {
+        webView.CoreWebView2.OpenDefaultDownloadDialog();
+    }
+}
+
+private void SetDefaultDownloadDialogPosition()
+{
+    const int defaultMargin = 20;
+    CoreWebView2DefaultDownloadDialogCornerAlignment cornerAlignment
+        = CoreWebView2DefaultDownloadDialogCornerAlignment.TopLeft;
+    System.Drawing.Point margin = new System.Drawing.Point(
+        defaultMargin, defaultMargin);
+    webView.CoreWebView2.DefaultDownloadDialogCornerAlignment =
+        cornerAlignment;
+    webView.CoreWebView2.DefaultDownloadDialogMargin = margin;
+}
+```
+
+# API Details
+
+
+```c#
+[v1_enum]
+/// The default download dialog can be aligned to any of the WebView corners
+/// by setting the `DefaultDownloadDialogCornerAlignment` property. The default
+/// position is top-right corner.
+typedef enum COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT {
+
+  /// Top-left corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT_TOP_LEFT,
+
+  /// Top-right corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT_TOP_RIGHT,
+
+  /// Bottom-left corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT_BOTTOM_LEFT,
+
+  /// Bottom-right corner of the WebView.
+  COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT_BOTTOM_RIGHT,
+
+} COREWEBVIEW2_DEFAULT_DOWNLOAD_DIALOG_CORNER_ALIGNMENT;
+
+[uuid(9139c04d-8f37-42ae-8b63-01940c34d22f), object, pointer_default(unique)]
+interface ICoreWebView2_6 : ICoreWebView2_5
+{
+  // Raised when the `IsDefaultDownloadDialogOpen` property changes.
+  HRESULT add_IsDefaultDownloadDialogOpenChanged(
+      [in] ICoreWebView2IsDefaultDownloadDialogOpenChangedEventHandler*
+          handler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_IsDefaultDownloadDialogOpenChanged`.
+  HRESULT remove_IsDefaultDownloadDialogOpenChanged(
+      [in] EventRegistrationToken token);
+
+  /// `TRUE` if the default download dialog is currently open. Toggling WebView
+  /// visibility also toggles default download dialog visibility.
+  [propget] HRESULT IsDefaultDownloadDialogOpen([out, retval] BOOL* value);
```

Comment:
Currently the table looks like this:

<html>
<body>
<!--StartFragment-->

WebViewController.IsVisible | IsDefault-Download-DialogOpen | IsDefault-DownloadDialog-OpenChanged raised? | Download dialog is on screen?
-- | -- | -- | --
changes from true to false | false | yes if it was open | no
changes from false to true | true if previously open | yes if previously open | yes if previously open

<!--EndFragment-->
</body>
</html>

Reply:
Seems strange as is. Let's change the behavior so that IsOpen property doesn't change when the visibility of the WebView (alone) changes. So it can be 'open' but not 'visible' if the webview isn't visible.